cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterContractCatalogController.java

```diff
@@ -0,0 +1,127 @@
+
+
+public class WaterContractCatalogController implements Handler {
+    private final Logger LOGGER = Logger.getLogger(WaterContractCatalogController.class.getName());
+    public static final String TAG = "Water Contracts";
+    private static final String WATER_USER = "water-user";
+    private final MetricRegistry metrics;
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    public WaterContractCatalogController(MetricRegistry metrics) {
+        this.metrics = metrics;
+    }
+
+    @NotNull
+    protected WaterContractDao getContractDao(DSLContext dsl) {
+        return new WaterContractDao(dsl);
+    }
+
+    @OpenApi(
+        pathParams = {
+            @OpenApiParam(name = OFFICE, description = "Specifies the"
+                    + " office that the contract is associated with.", required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "Specifies the project id of the contract.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "Specifies the water user of the contract.",
+                    required = true),
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_200,
+                content = {
+                    @OpenApiContent(from = WaterUserContract.class, type = Formats.JSONV1),
+                    @OpenApiContent(from = WaterUserContract.class, type = Formats.JSON)
+                }),
+            @OpenApiResponse(status = "404", description = "The provided combination of parameters"
+                    + " did not find any contracts."),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Return all water contracts",
+        path = "/projects/{office}/{project-id}/water-users/{water-user}/contracts",
+        method = HttpMethod.GET,
+        tags = {TAG}
+    )
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            final String office = ctx.queryParam(OFFICE);
+            final String name = ctx.queryParam(NAME);
+            final String locationId = ctx.pathParam(PROJECT_ID);
+            DSLContext dsl = getDslContext(ctx);
+            String result;
+            CwmsId projectLocation = new CwmsId.Builder().withOfficeId(office).withName(locationId).build();
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, WaterUserContract.class);
+            ctx.contentType(contentType.toString());
+            WaterContractDao contractDao = getContractDao(dsl);
+            List<WaterUserContract> contracts = contractDao.getAllWaterContracts(projectLocation, name);
+
+            if (contracts.isEmpty()) {
```

Comment:
Catalog or getAll calls shouldn't return 404 unless the required parameters do not exist since they are acting as filters. In this case the project-id and office-id are required and the db call will throw an exception if they do not exist (and the Javalin exception handlers will properly handle the 404 return code for you). The underlying database query doesn't require the water user name to not be null, but it will never return anything if it is and doesn't support any masking so it is basically a required field. You could make a db ticket to handle this (either start supporting masks or check to make sure the water user exists) or handle it in CDA by checking existence before executing the query and returning the 404 in that case. Either way the isEmpty() shouldn't be the qualifier for the 404.

Reply:
Added a Water User check before calling the contract DAO, no user results in 404